### PR TITLE
Prevents A Non-Admin Respawn Antag Wraith from Spawning If A Wraith is Already Alive

### PR DESCRIPTION
**Pull request by Wisemonster**
[BALANCE] [GAMEMODES]

## About the PR 
This prevents another wraith from being chosen as the random event antag if a wraith is still alive. This does not affect the admin event.

## Why's this needed? 
Multiple wraiths cause issues, both for the crew having to deal with multiple wraiths, and for other wraiths trying to absorb bodies.

### CODE REVIEW

File: code/modules/events/antag_ghost_respawn.dm [hunk 1]
```diff
@@ -44,6 +44,12 @@
 				return
 
 			src.antagonist_type = pick(list("Blob", "Hunter", "Werewolf", "Wizard", "Wraith", "Wrestler", "Wrestler_Doodle", "Vampire", "Changeling", "Flockmind"))
+			for(var/mob/wraith/W in ROLE_WRAITH)
+				if(W.deaths < 2)
+					src.antagonist_type -= list("Wraith")
+					src.antagonist_type = pick(list())
+				if(isnull(W.deaths))
+					return
```
Comment (Tarmunora) [inline code/modules/events/antag_ghost_respawn.dm:52]:
```suggestion
					continue
```
?

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/events/antag_ghost_respawn.dm [hunk 2]
```diff
@@ -44,6 +44,12 @@
 				return
 
 			src.antagonist_type = pick(list("Blob", "Hunter", "Werewolf", "Wizard", "Wraith", "Wrestler", "Wrestler_Doodle", "Vampire", "Changeling", "Flockmind"))
+			for(var/mob/wraith/W in ROLE_WRAITH)
```
Comment (Tarmunora) [inline code/modules/events/antag_ghost_respawn.dm:47]: also this loop condition does not look right. I think you're iterating over the wrong thing

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/modules/events/antag_ghost_respawn.dm (hunk 2 above)
Comment (Wisemonster) [reply code/modules/events/antag_ghost_respawn.dm:47]: I'm trying to have it check that out of the current wraiths around, which ones have died less than twice. I'm unsure if there's something else that checks if a wraith is dead-dead.

Comment (Wisemonster) [reply code/modules/events/antag_ghost_respawn.dm:47]: I think I see the problem, gonna replace the ROLE_WRAITH with the list for selected traitors.

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**